runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/inventory/Inventory.java

```diff
@@ -452,6 +454,19 @@ public static boolean useItemOnItem(String itemName1, String itemName2) {
         return true;
     }
 
+    public static boolean useItemOnNpc(String itemName, String npcName) {
+        if (Rs2Bank.isOpen()) return false;
+        Microbot.status = "Use inventory item " + itemName + " on " + npcName;
+        Widget item1 = findItem(itemName);
+        NPC item2 = getNpc(npcName);
```

Comment:
you can pass item2 in the Rs2Npc.interact(item2, "use")

this way you don't fetch the npc twice